app/src/test/java/org/simple/clinic/home/patients/PatientsScreenControllerTest.kt

```diff
@@ -446,8 +447,26 @@ class PatientsScreenControllerTest {
   }
 
   @Test
-  fun `when the user clicks scan card id button, open ScanSimpleIdCardScreen`() {
+  fun `when the user clicks scan card id button, request for camera permissions`() {
     uiEvents.onNext(ScanCardIdButtonClicked)
-    verify(screen).openScanSimpleIdCardScreen()
+    verify(screen).requestCameraPermissions()
+  }
+
+  @Test
+  @Parameters(value = [
+    "GRANTED|true",
+    "DENIED|false",
+    "NEVER_ASK_AGAIN|false"
+  ])
```

Comment:
For non primitive parameters, can we create the values dynamically instead of hardcoding them? Refactoring them becomes very difficult especially if the class signature is changing. That's one huge reason why Rakshak and I couldn't remove `UserInputDateValidator.Result` entirely from the codebase.

Reply:
Good point. I have made this change.